contracts/AaveV3Lender.sol

```diff
@@ -197,43 +203,80 @@ contract AaveV3Lender is BaseTokenizedStrategy, UniswapV3Swapper {
         for (uint256 i = 0; i < rewardsList.length; ++i) {
             token = rewardsList[i];
 
-            if (token == asset || dontSell[token]) {
+            if (token == asset) {
                 continue;
             } else {
-                _swapFrom(
-                    token,
-                    asset,
-                    ERC20(token).balanceOf(address(this)),
-                    0
-                );
+                uint256 balance = ERC20(token).balanceOf(address(this));
+
+                if (balance > minAmountToSellMapping[token]) {
+                    _swapFrom(token, asset, balance, 0);
+                }
             }
         }
     }
 
+    /**
+     * @notice Gets the max amount of `asset` that can be withdrawn.
+     * @dev Defaults to an unlimited amount for any address. But can
+     * be overriden by strategists.
+     *
+     * This function will be called before any withdraw or redeem to enforce
+     * any limits desired by the strategist. This can be used for illiquid
+     * or sandwhichable strategies. It should never be lower than `totalIdle`.
+     *
+     *   EX:
+     *       return TokenIzedStrategy.totalIdle();
+     *
+     * This does not need to take into account the `_owner`'s share balance
+     * or conversion rates from shares to assets.
+     *
+     * @param . The address that is withdrawing from the strategy.
+     * @return . The avialable amount that can be withdrawn in terms of `asset`
+     */
+    function availableWithdrawLimit(
+        address /*_owner*/
+    ) public view override returns (uint256) {
+        return ERC20(asset).balanceOf(address(aToken));
```

Comment:
Would it make sense to add idle asset from strategy? Like: `return ERC20(asset).balanceOf(address(aToken)) + ERC20(asset).balanceOf(address(this));`

Reply:
Yes, thats a very good point. Would be needed if the strategy is shutdown. Which would likely be happening when the withdraw limits are applicable

Reply:
added, with a totalIdle() check